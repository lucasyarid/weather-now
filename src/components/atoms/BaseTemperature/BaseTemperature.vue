<template>
  <div class="BaseTemperature" :class="BaseTemperatureColor">
    {{ temperature }}<span class="BaseTemperature__scale">Â°</span>
  </div>
</template>

<script>
export default {
  name: "BaseTemperature",
  props: {
    temperature: {
      type: Number,
      required: true
    },
    scale: {
      type: String,
      default: "Celsius"
    }
  },
  computed: {
    BaseTemperatureColor() {
      let color;
      if (this.temperature <= 5) color = "blue";
      if (this.temperature > 5 && this.temperature <= 25) color = "orange";
      if (this.temperature > 25) color = "red";

      return `BaseTemperature--${color}`;
    }
  }
};
</script>

<style lang="scss" scoped>
.BaseTemperature {
  font-size: 96px;
  display: flex;

  &--blue {
    color: $color-blue;
  }
  &--orange {
    color: $color-orange;
  }
  &--red {
    color: $color-red;
  }

  &__scale {
    font-size: 52px;
    transform: translateY(15%);
  }
}
</style>
