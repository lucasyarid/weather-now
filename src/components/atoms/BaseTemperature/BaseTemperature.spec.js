import { shallowMount } from "@vue/test-utils";
import BaseTemperature from "./BaseTemperature";

describe("BaseTemperature", () => {
  const WRAPPER = shallowMount(BaseTemperature, {
    propsData: {
      temperature: -15
    }
  });

  test("Componente existe", () => {
    expect(WRAPPER.exists()).toBe(true);
  });

  test("Cor da temperatura", () => {
    WRAPPER.setProps({ temperature: -15 });
    expect(WRAPPER.vm.BaseTemperatureColor).toBe("BaseTemperature--blue");

    WRAPPER.setProps({ temperature: 10 });
    expect(WRAPPER.vm.BaseTemperatureColor).toBe("BaseTemperature--orange");

    WRAPPER.setProps({ temperature: 30 });
    expect(WRAPPER.vm.BaseTemperatureColor).toBe("BaseTemperature--red");
  });

  test("Componente nÃ£o alterado", () => {
    expect(WRAPPER.html()).toMatchSnapshot();
  });
});
