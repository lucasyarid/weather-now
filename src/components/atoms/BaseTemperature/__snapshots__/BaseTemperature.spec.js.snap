// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseTemperature Componente não alterado 1`] = `
"<div class=\\"BaseTemperature BaseTemperature--red\\">
  30<span class=\\"BaseTemperature__scale\\">°</span></div>"
`;
